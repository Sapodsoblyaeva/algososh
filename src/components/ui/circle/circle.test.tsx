import { render, screen } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("circle test", () => {
  it("circle without letter", () => {
    render(<Circle />);

    const circle = screen.getByTestId("circle");

    expect(circle).not.toHaveTextContent("c");
    expect(circle).toMatchSnapshot();
  });

  it("circle with letter", () => {
    render(<Circle letter="c" />);

    const circle = screen.getByTestId("circle");

    expect(circle).toHaveTextContent("c");
    expect(circle).toMatchSnapshot();
  });

  it("circle with head", () => {
    render(<Circle head="head" />);

    const circleHead = screen.getByTestId("circle-head");

    expect(circleHead).toBeInTheDocument();
    expect(circleHead).toHaveTextContent("head");
    expect(circleHead).toMatchSnapshot();
  });

  it("circle with element as head", async () => {
    render(<Circle head={<Circle />} />);

    const circleHeads = await screen.findAllByTestId("circle");

    expect(circleHeads.length).toBe(2);
    expect(circleHeads).toEqual(circleHeads);
    expect(circleHeads).toMatchSnapshot();
  });

  it("circle with tail", () => {
    render(<Circle tail="tail" />);

    const circleTail = screen.getByTestId("circle-tail");

    expect(circleTail).toBeInTheDocument();
    expect(circleTail).toHaveTextContent("tail");
    expect(circleTail).toMatchSnapshot();
  });

  it("circle with element as tail", async () => {
    render(<Circle tail={<Circle />} />);

    const circleTails = await screen.findAllByTestId("circle");

    expect(circleTails.length).toBe(2);
    expect(circleTails).toEqual(circleTails);
    expect(circleTails).toMatchSnapshot();
  });

  it("circle with index", () => {
    render(<Circle index={2} />);

    const circleIndexElem = screen.getByTestId("circle-index");

    expect(circleIndexElem).toBeInTheDocument();
    expect(circleIndexElem).toHaveTextContent("2");
    expect(circleIndexElem).toMatchSnapshot();
  });

  it("circle with Prop isSmall", () => {
    render(<Circle isSmall={true} />);

    const circleSize = screen.getByTestId("circle-view");

    expect(circleSize).toHaveClass("small");
    expect(circleSize).toMatchSnapshot();
  });

  it("circle with default state", () => {
    render(<Circle state={ElementStates.Default} />);

    const circle = screen.getByTestId("circle-view");

    expect(circle).toHaveClass(ElementStates.Default);
    expect(circle).toMatchSnapshot();
  });

  it("circle with modified state", () => {
    render(<Circle state={ElementStates.Modified} />);

    const circle = screen.getByTestId("circle-view");

    expect(circle).toHaveStyle({border: "4px solid var(--modified-color"});
    expect(circle).toHaveClass(ElementStates.Modified);
    expect(circle).toMatchSnapshot();
  });

  it("circle with changing state", () => {
    render(<Circle state={ElementStates.Changing} />);

    const circle = screen.getByTestId("circle-view");

    expect(circle).toHaveStyle({border: "4px solid var(--changing-color)"});
    expect(circle).toMatchSnapshot();
  });
});
