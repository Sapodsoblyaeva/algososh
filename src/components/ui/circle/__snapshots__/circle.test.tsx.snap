// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle test circle with Prop isSmall 1`] = `
<div
  class="circle  small default"
  data-testid="circle-view"
>
  <p
    class="text text_type_circle text_color_input letter"
  />
</div>
`;

exports[`circle test circle with changing state 1`] = `
<div
  class="circle   changing"
  data-testid="circle-view"
>
  <p
    class="text text_type_circle text_color_input letter"
  />
</div>
`;

exports[`circle test circle with default state 1`] = `
<div
  class="circle   default"
  data-testid="circle-view"
>
  <p
    class="text text_type_circle text_color_input letter"
  />
</div>
`;

exports[`circle test circle with element as head 1`] = `
Array [
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    >
      <div
        class="content "
        data-testid="circle"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
          data-testid="circle-head"
        />
        <div
          class="circle   default"
          data-testid="circle-view"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
          data-testid="circle-index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
          data-testid="circle-tail"
        />
      </div>
    </div>
    <div
      class="circle   default"
      data-testid="circle-view"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
      data-testid="circle-index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>,
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle-view"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
      data-testid="circle-index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>,
]
`;

exports[`circle test circle with element as tail 1`] = `
Array [
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle-view"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
      data-testid="circle-index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    >
      <div
        class="content "
        data-testid="circle"
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
          data-testid="circle-head"
        />
        <div
          class="circle   default"
          data-testid="circle-view"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
          data-testid="circle-index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
          data-testid="circle-tail"
        />
      </div>
    </div>
  </div>,
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle-view"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
      data-testid="circle-index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>,
]
`;

exports[`circle test circle with head 1`] = `
<div
  class="text text_type_input text_color_input mb-4 absolute head string"
  data-testid="circle-head"
>
  head
</div>
`;

exports[`circle test circle with index 1`] = `
<p
  class="text text_type_input text_color_input mt-4 absolute index"
  data-testid="circle-index"
>
  2
</p>
`;

exports[`circle test circle with letter 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    class="circle   default"
    data-testid="circle-view"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      c
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`circle test circle with modified state 1`] = `
<div
  class="circle   modified"
  data-testid="circle-view"
>
  <p
    class="text text_type_circle text_color_input letter"
  />
</div>
`;

exports[`circle test circle with tail 1`] = `
<div
  class="text text_type_input text_color_input mt-4 absolute tail30 string"
  data-testid="circle-tail"
>
  tail
</div>
`;

exports[`circle test circle without letter 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    class="circle   default"
    data-testid="circle-view"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;
